test/prov_recurse_config_test.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/evp.h>
+#include "testutil.h"
+
+static char *configfile = NULL;
+
+/*
+ * Load a provider config with a recursive section and check
+ * for errors after the fact to ensure we caught and aborted the
+ * load
+ */
+
+static int errfound = 0;
+
+static int err_cb(const char *str, size_t len, void *u)
+{
+    errfound = 1;
+    return 1;
+}
+
+static int test_recursive_config(void)
+{
+    OSSL_LIB_CTX *ctx = OSSL_LIB_CTX_new();
+    int testresult = 0;
+
+    if (!TEST_ptr(configfile))
+        goto err;
+
+    if (!TEST_ptr(ctx))
+        goto err;
+
+    if (!TEST_true(OSSL_LIB_CTX_load_config(ctx, configfile)))
+        goto err;
+
+    ERR_print_errors_cb(err_cb, NULL);
```

Comment:
I'd rather use ERR_peek_error() or ERR_peek_last_error() to check whether there is a concrete error returned.

Reply:
As long as we're sure that the the parser error is at the top of the stack (which I think should always be the case), sure